File: JavaApp/src/main/java/com/charlezz/javaapp/feature/local/UserViewModel.java
package com.charlezz.javaapp.feature.local;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;
import androidx.lifecycle.ViewModel;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

import javax.inject.Inject;

public class UserViewModel extends ViewModel {

    private LiveData<PagedList<User>> users;
    private MediatorLiveData<Boolean> isLoaded = new MediatorLiveData<>();

    @Inject
    public UserViewModel(UserDao userDao){
        isLoaded.setValue(false);
        users = new LivePagedListBuilder<>(userDao.getUsers(),20).build();
        isLoaded.addSource(users, people -> isLoaded.setValue(people!=null && !people.isEmpty()));
    }

    public LiveData<PagedList<User>> getUsers() {
        return users;
    }

    public LiveData<Boolean> isLoaded() {
        return isLoaded;
    }

}

File: KotlinApp/src/main/java/com/charlezz/finalarchitecture/feature/local/UserViewModel.kt
package com.charlezz.finalarchitecture.feature.local

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import androidx.paging.LivePagedListBuilder
import androidx.paging.PagedList
import javax.inject.Inject


class UserViewModel @Inject constructor(personDao: UserDao) : ViewModel() {
    val persons:LiveData<PagedList<User>> = LivePagedListBuilder(personDao.getPersonSource(), 10).build()
    val isLoaded = Transformations.switchMap(persons){
        input: PagedList<User>? ->
        MutableLiveData<Boolean>().apply { value = (input!=null&&input.size!=0) }
    }
}